test/Infrastructure.IntegrationTest/AdminConsole/Repositories/ProviderUserRepositoryTests.cs

```diff
@@ -0,0 +1,151 @@
+        {
+            ProviderId = provider.Id,
+            UserId = user.Id,
+            Status = ProviderUserStatusType.Confirmed,
+            Type = ProviderUserType.ProviderAdmin
+        });
+
+        var providerOrganizationWithSso = await providerOrganizationRepository.CreateAsync(new ProviderOrganization
+        {
+            ProviderId = provider.Id,
+            OrganizationId = organizationWithSso.Id
+        });
+
+        var providerOrganizationWithoutSso = await providerOrganizationRepository.CreateAsync(new ProviderOrganization
+        {
+            ProviderId = provider.Id,
+            OrganizationId = organizationWithoutSso.Id
+        });
+
+        // Create SSO configuration for first organization only
+        var serializedSsoConfigData = new SsoConfigurationData
+        {
+            MemberDecryptionType = MemberDecryptionType.KeyConnector,
+            KeyConnectorUrl = "https://keyconnector.example.com"
+        }.Serialize();
+
+        var ssoConfig = await ssoConfigRepository.CreateAsync(new SsoConfig
+        {
+            OrganizationId = organizationWithSso.Id,
+            Enabled = true,
+            Data = serializedSsoConfigData
+        });
+        var results = (await providerUserRepository.GetManyOrganizationDetailsByUserAsync(user.Id, ProviderUserStatusType.Confirmed)).ToList();
+
+        Assert.Equal(2, results.Count);
+
+        var orgWithSsoDetails = results.Single(r => r.OrganizationId == organizationWithSso.Id);
+        var orgWithoutSsoDetails = results.Single(r => r.OrganizationId == organizationWithoutSso.Id);
+
+        // Verify all properties for both organizations
+        AssertProviderOrganizationDetails(orgWithSsoDetails, organizationWithSso, user, provider, providerUser);
+        AssertProviderOrganizationDetails(orgWithoutSsoDetails, organizationWithoutSso, user, provider, providerUser);
+
+        // Organization without SSO should have null SSO properties
+        Assert.Null(orgWithoutSsoDetails.SsoEnabled);
+        Assert.Null(orgWithoutSsoDetails.SsoConfig);
+
+        // Organization with SSO should have SSO properties populated
+        Assert.True(orgWithSsoDetails.SsoEnabled);
+        Assert.NotNull(orgWithSsoDetails.SsoConfig);
+        Assert.Equal(serializedSsoConfigData, orgWithSsoDetails.SsoConfig);
+
+        // Cleanup
+        await ssoConfigRepository.DeleteAsync(ssoConfig);
+        await providerOrganizationRepository.DeleteAsync(providerOrganizationWithSso);
+        await providerOrganizationRepository.DeleteAsync(providerOrganizationWithoutSso);
+        await providerUserRepository.DeleteAsync(providerUser);
+        await providerRepository.DeleteAsync(provider);
+        await organizationRepository.DeleteAsync(organizationWithSso);
+        await organizationRepository.DeleteAsync(organizationWithoutSso);
+        await userRepository.DeleteAsync(user);
+    }
+
+    private static void AssertProviderOrganizationDetails(
```

Comment:
Looking at this again, the test helpers create a very basic organization without many properties set. That means most properties would default to null or false, which is indistinguishable from a sproc that is missing a column.

I suggest using Autodata to create the organization object so that it's populated with identifiable data.

Reply:
I've added a private method that initializes all properties. To use `new Fixture()` I would have to add the `AutoFixture` package and it seemed to be out of scope here, I can add it later if you think its worth it.